Sources/Build/BuildPlan.swift

```diff
@@ -717,8 +724,13 @@ public final class SwiftTargetBuildDescription {
             // can construct the linker flags. In the future we will use a generated
             // code stub for the cases in which the linker doesn't support it, so that
             // we can rename the symbol unconditionally.
-            if buildParameters.linkerFlagsForRenamingMainFunction(of: target) != nil {
-                args += ["-Xfrontend", "-entry-point-function-name", "-Xfrontend", "\(target.c99name)_main"]
+            // No `-` for these flags because the set of Strings in driver.supportedFrontendFlags do
+            // not have a leading `-`
+            let flags: Set = ["Xfrontend", "entry-point-function-name"]
```

Comment:
Converting this to a set won't preserve the ordering, so I don't think the `Xfrontend` does anything here except to check that the driver supports any frontend flag.

Reply:
`-Xfrontend` is the general flag for passing frontend flags to the compiler driver.

Reply:
I only thought to make it a set in order to take advantage of the set intersection, that makes it easy to check if multiple flags are present in one go. So do you think this should just check for `entry-point-function-name`?

Reply:
Yes, I think that the question the BuildPlan should be asking of the toolchain is whether the Swift compiler supports renaming the entry point, and then the fact that that is implemented using the SwiftDriver capabilities seems like an implementation detail.

Reply:
Sometimes it good to generalize things, and so you might want one general method that takes a set but then another one that uses that to check for the specific flag.  In this case it would be great to only spell out the name of the flag once, in a computed property or a method that is specific for checking if custom entry points are supported, and then have the other code use that rather than repeat the name of the flag (as the tests do).